gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -962,4 +967,915 @@ public void testListDnsRecords() {
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.START_TIME));
+      DnsBatchResult<Page<ChangeRequest>> resultStatus = batch.listChangeRequests(ZONE1.name(),
+          Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+          Dns.ChangeRequestListOption.fields(ChangeRequestField.STATUS));
+      batch.submit();
+      assertEquals(3, Iterables.size(result.get().values()));
+      // error in options
+      try {
+        errorPageSize.get();
+        fail();
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(400, ex.code());
+        assertFalse(ex.retryable());
+      }
+      try {
+        errorPageNegative.get();
+        fail();
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(400, ex.code());
+        assertFalse(ex.retryable());
+      }
+      // sorting order
+      ImmutableList<ChangeRequest> ascending =
+          ImmutableList.copyOf(resultAscending.get().iterateAll());
+      ImmutableList<ChangeRequest> descending =
+          ImmutableList.copyOf(resultDescending.get().iterateAll());
+      int size = 3;
+      assertEquals(size, descending.size());
+      assertEquals(size, ascending.size());
+      for (int i = 0; i < size; i++) {
+        assertEquals(descending.get(i), ascending.get(size - i - 1));
+      }
+      // field options
+      change = Iterables.get(resultAdditions.get().values(), 1);
+      assertEquals(CHANGE_ADD_ZONE1.additions(), change.additions());
+      assertTrue(change.deletions().isEmpty());
+      assertNotNull(change.generatedId());
+      assertNull(change.startTimeMillis());
+      assertNull(change.status());
+      change = Iterables.get(resultDeletions.get().values(), 2);
+      assertTrue(change.additions().isEmpty());
+      assertNotNull(change.deletions());
+      assertNotNull(change.generatedId());
+      assertNull(change.startTimeMillis());
+      assertNull(change.status());
+      change = Iterables.get(resultId.get().values(), 1);
+      assertTrue(change.additions().isEmpty());
+      assertTrue(change.deletions().isEmpty());
+      assertNotNull(change.generatedId());
+      assertNull(change.startTimeMillis());
+      assertNull(change.status());
+      change = Iterables.get(resultTime.get().values(), 1);
+      assertTrue(change.additions().isEmpty());
+      assertTrue(change.deletions().isEmpty());
+      assertNotNull(change.generatedId());
+      assertNotNull(change.startTimeMillis());
+      assertNull(change.status());
+      change = Iterables.get(resultStatus.get().values(), 1);
+      assertTrue(change.additions().isEmpty());
+      assertTrue(change.deletions().isEmpty());
+      assertNotNull(change.generatedId());
+      assertNull(change.startTimeMillis());
+      assertEquals(ChangeRequest.Status.DONE, change.status());
+    } finally {
+      clear();
+    }
+  }
+
+  @Test
+  public void testListDnsRecordSetsBatch() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method testListDnsRecordSetsBatch() has an NPath complexity of 1025](https://www.codacy.com/app/mziccard/gcloud-java/file/2428842112/issues/source?bid=3123103&fileBranchId=3240468#l1707)